### PR TITLE
MWPW-143591 Fix Contextual Search Duplicates

### PR DESCRIPTION
**Pull request by meganthecoder**
* Allows feed config in contextual search authoring for customized card results
* Fixes duplicates in contextual search

Resolves: [MWPW-143591](https://jira.corp.adobe.com/browse/MWPW-143591)

**Test URLs:**
Milo:
- Before: https://main--milo--adobecom.hlx.page/drafts/methomas/contextual-search
- After: https://methomas-search-exclude--milo--adobecom.hlx.page/drafts/methomas/contextual-search

Bacom Blog
- Before: https://main--bacom-blog--adobecom.hlx.page/uk/drafts/methomas/?martech=off
- After: https://main--bacom-blog--adobecom.hlx.page/uk/drafts/methomas/?milolibs=methomas-search-exclude&martech=off

Blog:
- Before: https://main--blog--adobecom.hlx.live/
- After: https://main--blog--adobecom.hlx.live/?milolibs=methomas-search-exclude&martech=off

### CODE REVIEW

**Review by Brandon32 — APPROVED** [review]

**Review by JasonHowellSlavin — COMMENTED** [review]
Looks good. Are there any tests that should be updated?

**Comment by meganthecoder** [comment]
> Looks good. Are there any tests that should be updated?

Not that I can see. Unit tests pass and code coverage looks good.

File: libs/blocks/gnav/gnav-contextual-search.js [hunk 1]
```diff
@@ -52,33 +53,31 @@ function highlightTextElements(terms, elements) {
...
+  for (const article of articles) {
+    if (hits.length === LIMIT || signal.aborted) break;
+    const { category } = getArticleTaxonomy(article);
+    const text = [category, article.title, article.description].join(' ').toLowerCase();
+    if (terms.every((term) => text.includes(term))) {
+      hits.push(article);
     }
   }
+
+  if (!signal.aborted && (!articles.length || (hits.length !== LIMIT && !complete))) {
```
[the remark this answers is not in the document]
Comment (meganthecoder) [reply libs/blocks/gnav/gnav-contextual-search.js]: Sure, updated

**Review by JasonHowellSlavin — APPROVED** [review]

**Comment by rgclayton** [comment]
Looking good from what I can see.
I am having trouble reproducing the duplicate results. I saw your notes in the Jira ticket, but still wasn't able to see that in action. Any other tips?

**Comment by meganthecoder** [comment]
> Looking good from what I can see. I am having trouble reproducing the duplicate results. I saw your notes in the Jira ticket, but still wasn't able to see that in action. Any other tips?

@rgclayton It's easiest to see the duplicate results when you search something very specific. For example, on Adobe Blog, try "christian cowan". On bacom blog, I was testing with "adobe launches customer journey".

It happens most often on the first search, so if you're still not seeing it, refresh the page and search again.

**Comment by meganthecoder** [comment]
Note: I have the "critical" label because that's the priority of the ticket, added by the PM. The due date is Thurs, March 14.

**Review by rgclayton — APPROVED** [review]
Thanks for the details @meganthecoder.

Looks good on blog.adobe.com

**Comment by meganthecoder** [comment]
FYI There was a bug found where the search results take longer to show for certain queries. PM said that it wasn't a blocker for launch, so we'll create a ticket and follow up later.

**Pull request merged**